src/RoslynInsertionTool/RoslynInsertionTool/CoreXT.cs

```diff
@@ -157,30 +157,144 @@ public bool TryGetPackageVersion(PackageInfo packageInfo, out SemanticVersion ve
         public bool TryGetComponentByName(string componentName, out Component component)
         {
             component = null;
-            if (Components != null)
+
+            JObject componentDocument = GetJsonDocumentForComponent(componentName);
+
+            if (componentDocument == null)
             {
-                var componentJSON = Components["Components"][componentName];
-                if (componentJSON == null)
-                {
-                    return false;
-                }
-                var componentFilename = (string)componentJSON["fileName"];
-                var componentUri = new Uri((string)componentJSON["url"]);
-                component = new Component(componentName, componentFilename, componentUri);
-                return true;
+                return false;
             }
 
-            return false;
+            var componentJSON = componentDocument["Components"][componentName];
+            if (componentJSON == null)
+            {
+                return false;
+            }
+
+            var componentFilename = (string)componentJSON["fileName"];
+            var componentUri = new Uri((string)componentJSON["url"]);
+            component = new Component(componentName, componentFilename, componentUri);
+            return true;
         }
 
         public void UpdateComponent(Component component)
         {
-            if (Components != null)
+            var componentDocument = GetJsonDocumentForComponent(component.Name);
+
+            if (componentDocument != null)
             {
-                var componentJSON = Components["Components"][component.Name];
-                componentJSON["fileName"] = component.Filename.ToString();
+                var componentJSON = componentDocument["Components"][component.Name];
+                componentJSON["fileName"] = component.Filename;
                 componentJSON["url"] = component.Uri.ToString();
+
+                string componentFilePath = ComponentToFileMap[component.Name];
+                if(!dirtyFiles.Contains(componentFilePath))
```

Comment:
HashSet, so the .Contains() isn't necessary. (Add no-ops if it's already there.)

Reply:
Agreed. Will remove.